### PR TITLE
show minimap with same rotation as map

### PR DESCRIPTION
**Pull request by jmdejong**
Since the map is isometric the minimap rotation was always (at least) 45 degrees off.

These changes solve that

Also, the minimap class has been majorly refactored, pulling most of the transformation code in its own class.

### CODE REVIEW

File: horizons/gui/widgets/minimap.py [hunk 1]
```diff
@@ -51,14 +42,19 @@ def iter_minimap_points(location, world, island_color, water_color, area=None):
 
 	If `area` is set, it's supposed to be a part of `location`, that is to be
 	returned.
+
+	This function is not used anymore in the in-game minimap, but it is necessary
```
Comment (AndyMender) [inline horizons/gui/widgets/minimap.py]: Wording - was this function used by the in-game minimap before? If it's used only for the preview, perhaps the code could be moved there?

Comment (jmdejong) [reply horizons/gui/widgets/minimap.py]: I wanted to remove this function, and use the _MinimapTransform class for that but that would require some more changes to _MinimapTransform.iter_points.
It's probably still better to fix that first though.

**Comment by jmdejong** [comment]
I just reailized that this PR is still missing some things, mostly related to the "rotate minimap with normal map" setting. When this setting is off it should go back to the old minimap, but the corners are cut of.
I also want to try to make that setting on by default.

File: horizons/gui/widgets/minimap.py (hunk 1 above)
Comment (AndyMender) [reply horizons/gui/widgets/minimap.py]: Then I think it's good as it is right now. I think smaller incremental changes are the way to go :).

**Comment by AndyMender** [comment]
@jmdejong We are grateful to receive any and all contributions. If you feel some extra things should be added to this PR, I'll happily review the additions :).

**Comment by LinuxDonald** [comment]
@jmdejong can this pr get merged or do you want add more stuff to it?

**Pull request merged**